### PR TITLE
[PAPI-325] Token Attributes

### PR DESCRIPTION
**Pull request by mrtpain**
- Remove `token.IsLpt` property
- Persist Token Attributes during token creation
    - NonProvisional, Provisional, and Staking attributes automatically inserted

Most changes are test fixes.

Related To: https://github.com/Opdex/opdex-v1-db/pull/33

### CODE REVIEW

File: src/Opdex.Platform.Infrastructure.Abstractions/Data/Queries/Tokens/TokensCursor.cs [hunk 1]
```diff
@@ -142,7 +142,7 @@ private static bool TryDecodePointer(string encoded, out (string, ulong) pointer
 /// <summary>
 /// Filter for a tokens status whether it is an Opdex liquidity pool token or not.
```
Comment (drmathias) [inline src/Opdex.Platform.Infrastructure.Abstractions/Data/Queries/Tokens/TokensCursor.cs:143]: nitpick: Update comment

File: src/Opdex.Platform.Infrastructure.Abstractions/Data/Queries/Tokens/TokensCursor.cs [hunk 2]
```diff
@@ -154,10 +154,15 @@ public enum TokenProvisionalFilter
...
+    /// <summary>
+    /// Opdex mined, staking tokens
+    /// </summary>
+    Staking = 2,
+
     /// <summary>
     /// All tokens.
     /// </summary>
-    All = 2
+    All = 3
```
Comment (drmathias) [inline src/Opdex.Platform.Infrastructure.Abstractions/Data/Queries/Tokens/TokensCursor.cs:165]: Maybe this should include `NonProvisional` and `Security` as well, for flexibility?

**Review by drmathias — APPROVED** [review]

File: src/Opdex.Platform.Infrastructure.Abstractions/Data/Queries/Tokens/TokensCursor.cs (hunk 2 above)
Comment (mrtpain) [reply src/Opdex.Platform.Infrastructure.Abstractions/Data/Queries/Tokens/TokensCursor.cs:165]: Good catch, didn't realize NonProvisional wasn't an option there. Will go back through this. Anytime anything is marked as a security we will likely not return anything about it.

I'll keep it out for now but move `All` to 1 so we can append more options cleaner down the road if we decide.

File: src/Opdex.Platform.WebApi/Models/Requests/Wallets/AddressBalanceFilterParameters.cs [hunk 3]
```diff
@@ -24,10 +24,10 @@ public AddressBalanceFilterParameters()
     public IEnumerable<Address> Tokens { get; set; }
 
     /// <summary>
-    /// The type of token to filter by, either provisional or non-provisional.
+    /// The types of token to filter balances by.
     /// </summary>
     /// <example>Provisional</example>
-    public TokenAttributeFilter TokenType { get; set; }
+    public IEnumerable<TokenAttributeFilter> TokenAttributes { get; set; }
```
Comment (mrtpain) [inline src/Opdex.Platform.WebApi/Models/Requests/Wallets/AddressBalanceFilterParameters.cs:30]: OAS changes

File: src/Opdex.Platform.WebApi/Models/Requests/Tokens/TokenFilterParameters.cs [hunk 4]
```diff
@@ -22,10 +23,10 @@ public TokenFilterParameters()
     public TokenOrderByType OrderBy { get; set; }
 
     /// <summary>
-    /// The type of token to filter for, liquidity pool tokens or not.
+    /// The types of token to filter for.
     /// </summary>
     /// <example>Provisional</example>
-    public TokenAttributeFilter TokenType { get; set; }
+    public IEnumerable<TokenAttributeFilter> TokenAttributes { get; set; }
```
Comment (mrtpain) [inline src/Opdex.Platform.WebApi/Models/Requests/Tokens/TokenFilterParameters.cs:29]: OAS changes @drmathias incase you've done this controller already

**Pull request merged**